UnityProject/Assets/Scripts/Health/Living/LivingHealthBehaviour.cs

```diff
@@ -400,6 +405,26 @@ protected void CalculateOverallHealth()
 		CheckHealthAndUpdateConsciousState();
 	}
 
+	[Server]
+	protected void CalculateStun(){
+		if(StunDuration > 0)
```

Comment:
is there a particular reason why you chose Update() over a coroutine? Coroutines are widely used in our codebase and are the right thing to use when you want cooldowns

Reply:
Wasn't sure what to do here, saw how some systems used the UpdateManager's events for timers so I decided to use that here.